src/Symfony/Component/Console/Tests/Resources/BashCompletionTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Tests\Resources;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @group doot
+ */
+class BashCompletionTest extends TestCase
+{
+    protected static $completionScript;
+
+    public static function setUpBeforeClass(): void
+    {
+        self::$completionScript = realpath(__DIR__.'/../../Resources/completion.bash');
+    }
+
+    public function testHasCompletionFunction()
+    {
+        // This is more of a test of the test environment than a test of the
+        // bash script; still useful if some joker changes the CI image
+        $required = [
+            '__ltrim_colon_completions',
+            '_get_comp_words_by_ref',
+        ];
+        $missing = [];
+        $prereq_command = $this->getPrerequisiteLoader();
+        foreach ($required as $func) {
+            $commands = [
+                $prereq_command,
+                'type -t '.escapeshellarg($func),
+            ];
+            $bash_command = implode("\n", $commands);
+            $command = 'bash -e -c '.escapeshellarg($bash_command).' 2>&1';
+            exec($command, $output, $exit_val);
+            if (0 !== $exit_val) {
+                $missing[] = $func;
+            }
+        }
+        $this->assertEmpty($missing);
```

Comment:
I have a more debuggable output with this:
```suggestion
        $this->assertEquals([], $missing);
```